env_canada/ec_radar.py

```diff
@@ -225,7 +233,7 @@ async def _get_legend(self):
         async with ClientSession(raise_for_status=True) as session:
             response = await session.get(url=geomet_url, params=legend_params)
             legend_bytes = await response.read()
-        self.legend_image = Image.open(BytesIO(legend_bytes)).convert("RGB")
+        self.legend_image = await _image_open(legend_bytes, "RGB")
```

Comment:
I would suspect that accessing any of the properties below will do I/O to read the image if open doesn't already load the image from disk into memory.   If unsure either way the safest course is to do all the image operations in the executor. Maybe create a new function on this object that sets all the image properties and than run that function in the executor with something like run_in_executor(None, self._set_image...)

Reply:
If I ran `get_loop` in the executor it seems it would catch everything. https://github.com/michaeldavie/env_canada/blob/master/env_canada/ec_radar.py#L332

Going to try that. Would love second opinion.

Reply:
Seems reasonable to move as much of that into the executor as possible (the async specific code that does the http requests, can't be)

Ideally you want to minimize the number of executor jobs as well since there is a cost to switch back and forth between threads.

If you are unsure if the work is doing I/O, run it in the executor.

Reply:
I'll see if there are other calls that can be put into a single executor job... without totally refactoring.